community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/EntityUpdates.java

```diff
@@ -221,22 +225,19 @@ public void onProperty( int propertyId, Value value )
             int[] propertyIds = schema.getPropertyIds();
             if ( relevantBefore && !relevantAfter )
             {
-                indexUpdates.add( IndexEntryUpdate.remove(
-                        nodeId, indexKey, valuesBefore( propertyIds )
-                    ) );
+                indexUpdates.add( IndexEntryUpdate.remove( entityId, indexKey, valuesBefore( propertyIds )
+                ) );
             }
             else if ( !relevantBefore && relevantAfter )
             {
-                indexUpdates.add( IndexEntryUpdate.add(
-                        nodeId, indexKey, valuesAfter( propertyIds )
+                indexUpdates.add( IndexEntryUpdate.add( entityId, indexKey, valuesAfter( propertyIds )
                 ) );
             }
-            else if ( relevantBefore )
+            else if ( relevantBefore && relevantAfter )
```

Comment:
This is the constant-condition, by the way, because `relevantAfter` is always `true`, but this is not clear from the code, so it was added to the condition to be more descriptive about what case this particular clause was for, even though it's superfluous.

Reply:
@andreikoval Not sure I agree here, I feel like the logic is easier to follow like this.